types/artifact.go

```diff
@@ -82,24 +90,26 @@ type ArtifactInfo struct {
 // BlobInfo is stored in cache
 type BlobInfo struct {
 	SchemaVersion     int
-	Digest            string             `json:",omitempty"`
-	DiffID            string             `json:",omitempty"`
-	OS                *OS                `json:",omitempty"`
-	PackageInfos      []PackageInfo      `json:",omitempty"`
-	Applications      []Application      `json:",omitempty"`
-	Misconfigurations []Misconfiguration `json:",omitempty"`
-	OpaqueDirs        []string           `json:",omitempty"`
-	WhiteoutFiles     []string           `json:",omitempty"`
-	Size              int                `json:",omitempty"`
+	Digest            string                       `json:",omitempty"`
+	DiffID            string                       `json:",omitempty"`
+	OS                *OS                          `json:",omitempty"`
+	PackageInfos      []PackageInfo                `json:",omitempty"`
+	Applications      []Application                `json:",omitempty"`
+	Misconfigurations []Misconfiguration           `json:",omitempty"`
+	OpaqueDirs        []string                     `json:",omitempty"`
+	WhiteoutFiles     []string                     `json:",omitempty"`
+	Size              int                          `json:",omitempty"`
+	CustomResources   map[string]AnalyzerResources `json:",omitempty"`
 }
 
 // ArtifactDetail is generated by applying blobs
 type ArtifactDetail struct {
-	OS                *OS                `json:",omitempty"`
-	Packages          []Package          `json:",omitempty"`
-	Applications      []Application      `json:",omitempty"`
-	Misconfigurations []Misconfiguration `json:",omitempty"`
-	Size              int                `json:",omitempty"`
+	OS                *OS                          `json:",omitempty"`
+	Packages          []Package                    `json:",omitempty"`
+	Applications      []Application                `json:",omitempty"`
+	Misconfigurations []Misconfiguration           `json:",omitempty"`
+	Size              int                          `json:",omitempty"`
+	CustomResources   map[string]AnalyzerResources `json:",omitempty"`
```

Comment:
`CustomResources` is unnecessary for ArtifactDetail now.

Reply:
Yes, map is not needed. I have replaced it with array.